internal/hlsproxy/hlsproxy.go

```
@@ -0,0 +1,166 @@
+	if p.referer != "" {
+		req.Header.Set("Referer", p.referer)
+	}
+	if p.userAgent != "" {
+		req.Header.Set("User-Agent", p.userAgent)
+	}
+	resp, err := p.client.Do(req)
+	if err != nil {
+		http.Error(w, "upstream fetch failed", http.StatusBadGateway)
+		return
+	}
+	defer resp.Body.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `resp.Body.Close` is not checked

(errcheck)